text/0000-io-error-handling.md

```diff
@@ -0,0 +1,81 @@
+- Feature Name: my_awesome_io_error_handling
```

Comment:
This should be an amendment to the io RFC and as such does not deserve its own feature gate. I'll see if I can come up with something better.